client/src/game/world.ts

```diff
@@ -159,8 +206,8 @@ export class World {
     }
 
     addUnit(unit: UnitFragment) {
-        const u = Unit.from(unit, this.factions, this.ruleset)
         const region = this.getRegion(unit)
+        const u = Unit.from(unit, region, this.factions, this.ruleset)
         const structure = unit.structureNumber ? region.structures.find(x => x.num === unit.structureNumber) : null
 
         region.addUnit(u, structure)
```

Comment:
Calling `region.AddUnit` will set Unit's `region` property to the `region` units is beeing added into.